tests/test_schema.py

```diff
@@ -593,6 +594,35 @@ def _assert_migration_consistency(self, schema_text):
                 f'DDL text was:\n{ddl_text}'
             )
 
+    def _assert_migration_equivalence(self, migrations):
+        # Compare 2 schemas obtained by multiple-step migration to a
+        # single-step migration.
+
+        # Validate that the final schema state has consistent migration.
+        self._assert_migration_consistency(migrations[-1])
+
+        # Jump to final schema state in a single migration.
+        single_migration = self.load_schema(migrations[-1])
+
+        # Evolve a schema in a series of migrations.
+        multi_migration = self.load_schema(migrations[0])
+        for i, state in enumerate(migrations[1:]):
+            multi_migration = self.run_ddl(multi_migration, f'''
+                CREATE MIGRATION m{i} TO {{
+                    {state}
+                }};
+                COMMIT MIGRATION m{i};
+            ''', 'test')
+
+        diff = s_ddl.delta_modules(single_migration, multi_migration, ['test'])
```

Comment:
Is there a reason you're using `delta_modules` and not `delta_schema` here?

Reply:
The `delta_schema` didn't work (was throwing exceptions about getting a Module instead of str), so I figured that I can get a bunch of tests to work this way and worry about why `delta_schema` didn't work later.
You can try changing the code to use `delta_schema` and see the error. I'm not certain whether there's something in the way I initialize the schemas that's wrong or whether it's a bug proper.

Reply:
I don't see how what you're doing here differs from `_assert_migration_consistency` which uses `delta_schema` without an issue.

Reply:
For some of the migration tests I have I'd also like to test what happens to actual data in the process (like whether the types, props and links get renamed or dropped, etc.). This would probably be done in a separate test class. The question is whether there's any value in doing some schema validation via introspection queries for these tests in addition to validating how they handle data? Or would that introspection be redundant in light of already having these schema tests?

Reply:
> I don't see how what you're doing here differs from `_assert_migration_consistency` which uses `delta_schema` without an issue.

Yeah, I agree, at a glance I expected it to work, but it didn't. I can try poking at the error in more details to see if it's something relatively simple.

Reply:
Well, the tests are already useful as is, as far as I can tell. So I'll commit this batch and build more tests on top of it so that you can get access to this and look at them too.

Reply:
This test class is actually supposed to test the `GET MIGRATION` consistency.  If you want to do fine-grained migration testing, make a separate class please.

Reply:
> This test class is actually supposed to test the `GET MIGRATION` consistency. If you want to do fine-grained migration testing, make a separate class please.

Well, the main reason why I stuck my tests in this class was that in addition to testing incremental SDL migrations I _also_ wanted to validate the final state w.r.t. `GET MIGRATION` consistency. (The point is to validate that the final state is probably sane *and* that the multiple SDL migrations do actually result in that same final state as a single jump.)

My question about doing more fine-grained migration testing w.r.t. data is mainly about whether the introspection queries would be redundant or not given that step-by-step migration of the same SDL would already be tested separately.

Reply:
Schema dump is reliable enough, so you just need to compare the result of the step-by-step migration with the result of a direct load of the final step.